components/brave_rewards/core/BUILD.gn

```diff
@@ -0,0 +1,571 @@
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//third_party/protobuf/proto_library.gni")
+
+import("//brave/components/brave_rewards/core/config.gni")
+import("//build/buildflag_header.gni")
+
+if (is_android) {
+  import("//build/config/android/rules.gni")
+}
+
+buildflag_header("buildflags") {
+  header_dir = "brave/components/brave_rewards/core"
+  header = "buildflags.h"
+  flags = [
+    "GEMINI_OAUTH_STAGING_URL=\"$gemini_oauth_staging_url\"",
+    "BITFLYER_STAGING_URL=\"$bitflyer_staging_url\"",
+    "REWARDS_GRANT_DEV_ENDPOINT=\"$rewards_grant_dev_endpoint\"",
+    "REWARDS_GRANT_STAGING_ENDPOINT=\"$rewards_grant_staging_endpoint\"",
+    "REWARDS_GRANT_PROD_ENDPOINT=\"$rewards_grant_prod_endpoint\"",
+    "BITFLYER_CLIENT_ID=\"$bitflyer_client_id\"",
+    "BITFLYER_CLIENT_SECRET=\"$bitflyer_client_secret\"",
+    "BITFLYER_STAGING_CLIENT_ID=\"$bitflyer_staging_client_id\"",
+    "BITFLYER_STAGING_CLIENT_SECRET=\"$bitflyer_staging_client_secret\"",
+    "BITFLYER_STAGING_URL=\"$bitflyer_staging_url\"",
+    "GEMINI_API_URL=\"$gemini_api_url\"",
+    "GEMINI_API_STAGING_URL=\"$gemini_api_staging_url\"",
+    "GEMINI_OAUTH_URL=\"$gemini_oauth_url\"",
+    "GEMINI_OAUTH_STAGING_URL=\"$gemini_oauth_staging_url\"",
+    "GEMINI_WALLET_CLIENT_ID=\"$gemini_wallet_client_id\"",
+    "GEMINI_WALLET_CLIENT_SECRET=\"$gemini_wallet_client_secret\"",
+    "GEMINI_WALLET_STAGING_CLIENT_ID=\"$gemini_wallet_staging_client_id\"",
+    "GEMINI_WALLET_STAGING_CLIENT_SECRET=\"$gemini_wallet_staging_client_secret\"",
+    "UPHOLD_CLIENT_ID=\"$uphold_client_id\"",
+    "UPHOLD_CLIENT_SECRET=\"$uphold_client_secret\"",
+    "UPHOLD_STAGING_CLIENT_ID=\"$uphold_staging_client_id\"",
+    "UPHOLD_STAGING_CLIENT_SECRET=\"$uphold_staging_client_secret\"",
+  ]
+}
+
+group("core") {
+  public_deps = [ ":ledger" ]
+}
+
+source_set("headers") {
+  visibility = [
+    ":*",
+    "//chrome/browser:browser",
+
+    # add this only when ads is in the same process
+    # "//brave/vendor/bat-native-ads",
+    "//brave/browser/extensions:extensions",
+    "//brave/browser/ui:ui",
+    "//brave/components/brave_rewards/browser:browser",
+    "//brave/components/brave_rewards/browser:testutil",
+    "//brave/vendor/brave-ios:rewards_db",
+  ]
+
+  sources = [
+    "global_constants.h",
+    "ledger.h",
+    "ledger_client.h",
+    "ledger_database.cc",
+    "ledger_database.h",
+    "mojom_structs.h",
+    "option_keys.h",
+  ]
+
+  deps = [ "//sql:sql" ]
+
+  public_deps = [
+    ":buildflags",
+    "//brave/components/brave_rewards/common/mojom:ledger_interfaces",
+    "//brave/components/brave_rewards/common/mojom:ledger_types",
+  ]
+}
+
+proto_library("publishers_proto") {
+  sources = [
+    "publisher/protos/channel_response.proto",
+    "publisher/protos/publisher_prefix_list.proto",
+  ]
+}
+
+source_set("ledger") {
```

Comment:
I think we should use static_library for one of targets here.

Reply:
Are you suggesting this as an improvement over the existing build config? The current `bat-native-ledger` build files just use `source_set`.

Reply:
I've updated this target to use `static_library`. Does that look better?